en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/faq.md

```diff
@@ -0,0 +1,177 @@
+## Functionality
+
+### What are the application layer protocols supported by WSO2 Choreo Connect ?
+
+HTTP/1.1, HTTP/2
+
+### What are the API types supported by WSO2 Choreo Connect ?
+
+REST, SOAP, GraphQL, WebSocket
+
+### I want to do a custom validation for all the API requests in addition to the default validations. How can I do that?
+
+You can write a [custom filter]({{base_path}}/deploy-and-publish/deploy-on-gateway/choreo-connect/extensions/custom-filters/) in the Enforcer component or use a separate [interceptor service]({{base_path}}/deploy-and-publish/deploy-on-gateway/choreo-connect/message-transformation/interceptor-microservice/interceptor-microservice/). Also, you can perform custom request validation using the [Open Policy Agent (OPA)]({{base_path}}/design/api-security/opa-validation/custom-opa-policy-for-choreo-connect/).
+
+### Can I perform API request/response transformations?
+
+Yes, use a separate [interceptor service]({{base_path}}/deploy-and-publish/deploy-on-gateway/choreo-connect/message-transformation/interceptor-microservice/interceptor-microservice/). Also API policies can be used to perform simple transformations like adding/removing headers, rewriting paths, adding query params, changing HTTP methods, etc.
+
+## Security
+
+### What are the different methods available for API authentication?
+
+Choreo Connect supports several API authentication mechanisms including OAuth 2.0. Refer to the [API authentication documentation]({{base_path}}/deploy-and-publish/deploy-on-gateway/choreo-connect/security/api-authentication/api-authentication/) for more information.
+
+### Can I use a custom authorization header?
+
+Yes, you can configure a custom authorization header name in Choreo Connect. Refer to [Custom Authorization Header]({{base_path}}/deploy-and-publish/deploy-on-gateway/choreo-connect/security/use-a-custom-authorization-header/) for more information.
+
+### Can I disable API security for a specific API deployed in WSO2 Choreo Connect?
+
+Yes, security can be disabled at API, resource and operation level. Refer to [Disabling Security]({{base_path}}/deploy-and-publish/deploy-on-gateway/choreo-connect/security/api-authentication/disabling-security/) for more information.
+
+### What are the different methods of API authorization supported by WSO2 Choreo Connect?
+
+Scope validation, subscription validation, custom claims mapping and Open Policy Agent (OPA) validation. Refer to [API Authorization]({{base_path}}/deploy-and-publish/deploy-on-gateway/choreo-connect/security/api-authorization/scope-validation/) for more information.
+
+## Troubleshooting
+
+### Why am I getting Cross-Origin Resource Sharing (CORS) errors?
+
+It could be due to an invalid CORS configuration in the router component. Also another possibility is that the API is not deployed at all. Since the preflight request is usually sent prior to the actual request, users would see a CORS failure. Refer to the [CORS configuration] for more information.
```

Comment:
```suggestion
It could be due to an invalid CORS configuration in the Router component. Also, another possibility is that the API is not deployed at all. Since the preflight request is usually sent prior to the actual request, users would see a CORS failure. Refer to the [CORS configuration] for more information.
```